src/frontend/screens/Settings/MapSettings/OfflineAreas.tsx

```diff
@@ -0,0 +1,179 @@
+import { BottomSheetModalMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
+import * as React from "react";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import { ScrollView, StyleSheet } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+
+import { RED } from "../../../lib/styles";
+import {
+  BottomSheetContent,
+  BottomSheetModal,
+  useBottomSheetModal,
+} from "../../../sharedComponents/BottomSheetModal";
+import Button from "../../../sharedComponents/Button";
+import HeaderTitle from "../../../sharedComponents/HeaderTitle";
+import { ErrorIcon } from "../../../sharedComponents/icons";
+import Loading from "../../../sharedComponents/Loading";
+import { OfflineAreaCard } from "../../../sharedComponents/OfflineAreaCard";
+
+const m = defineMessages({
+  title: {
+    id: "screens.Settings.MapSettings.OfflineArea",
+    defaultMessage: "Offline Areas",
+  },
+  removeMap: {
+    id: "screens.Settings.MapSettings.removeMap",
+    defaultMessage: "Remove Map",
+  },
+  cancel: {
+    id: "screens.Settings.MapSettings.cancel",
+    defaultMessage: "Cancel",
+  },
+  clearDiagrams: {
+    id: "screens.Settings.MapSettings.clearDiagrams",
+    defaultMessage: "Are you sure you want to Clear diagrams",
+  },
+  subtitle: {
+    id: "screens.Settings.MapSettings.subtitle",
+    defaultMessage:
+      "This map and offline areas attached to it will be deleted. This cannot be undone",
+  },
+});
+
+interface OfflineArea {
+  id: string;
+  title: string;
+  zoomLevel: number;
+  levelOfDetail: string;
+}
+
+export const OfflineAreas: NavigationStackScreenComponent = ({
+  navigation,
+}) => {
+  const bgMapId = React.useRef("");
+  const { formatMessage: t } = useIntl();
+
+  const { closeSheet, isOpen, openSheet, sheetRef } = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  const [listOfflineAreas, setListOfflineAreas] = React.useState<
+    OfflineArea[]
+  >();
+
+  const { getParam } = navigation;
+
+  React.useEffect(() => {
+    //Typesafety?
+    bgMapId.current = getParam("mapId", "");
```

Comment:
context for this comment? is TS showing some kind of warning?

Reply:
No warning, just flagging that there is no typesafety and annoying complicated given that it is getting the nav item comes from the parent

Reply:
ah got it. yeah there's a way to make it more type-safe but not worth looking into for this (think it's improved in more recent version of react navigation)

Reply:
yeah its much easier in react nav 6 (i believe they wrote nav 6 in ts)...right now its a whole bunch of boillerplate code and it makes it really confusing because it looks like it is the params for the parent component (because you cant use the hook to pass params)